lib/pgBackRest/Common/Ini.pm

```diff
@@ -297,306 +391,249 @@ sub iniSave
-sub boolGet
-{
-    my $self = shift;
-    my $strSection = shift;
-    my $strValue = shift;
-    my $strSubValue = shift;
-    my $bRequired = shift;
-    my $bDefault = shift;
-
-    return $self->get($strSection, $strValue, $strSubValue, $bRequired,
-                      defined($bDefault) ? ($bDefault ? INI_TRUE : INI_FALSE) : undef) ? true : false;
-}
+sub boolGet {
+    return shift->get(shift, shift, shift, shift, defined($_[0]) ? (shift() ? INI_TRUE : INI_FALSE) : undef) ? true : false}
 
 ####################################################################################################################################
-# numericGet
-#
-# Get a numeric value.
+# numericGet() - get a numeric value.
 ####################################################################################################################################
-sub numericGet
-{
-    my $self = shift;
-    my $strSection = shift;
-    my $strValue = shift;
-    my $strSubValue = shift;
-    my $bRequired = shift;
-    my $nDefault = shift;
-
-    return $self->get($strSection, $strValue, $strSubValue, $bRequired,
-                      defined($nDefault) ? $nDefault + 0 : undef) + 0;
-}
+sub numericGet {return shift->get(shift, shift, shift, shift, defined($_[0]) ? shift() + 0 : undef) + 0}
 
 ####################################################################################################################################
-# set
-#
-# Set a value.
+# set - set a value.
 ####################################################################################################################################
 sub set
 {
     my $self = shift;
     my $strSection = shift;
     my $strKey = shift;
     my $strSubKey = shift;
-    my $strValue = shift;
+    my $oValue = shift;
+
+    # Parameter constraints
+    if (!(defined($strSection) && defined($strKey)))
+    {
+        confess &log(ASSERT, 'strSection and strKey are required');
+    }
 
-    my $oContent = $self->{oContent};
+    my $oCurrentValue;
 
     if (defined($strSubKey))
     {
-        ${$oContent}{$strSection}{$strKey}{$strSubKey} = $strValue;
+        $oCurrentValue = \$self->{oContent}{$strSection}{$strKey}{$strSubKey};
     }
     else
     {
-        ${$oContent}{$strSection}{$strKey} = $strValue;
+        $oCurrentValue = \$self->{oContent}{$strSection}{$strKey};
     }
-}
 
-####################################################################################################################################
-# boolSet
-#
-# Set a boolean value.
-####################################################################################################################################
-sub boolSet
-{
-    my $self = shift;
-    my $strSection = shift;
-    my $strKey = shift;
-    my $strSubKey = shift;
-    my $bValue = shift;
+    if (!defined($$oCurrentValue) ||
+        defined($oCurrentValue) != defined($oValue) ||
```

Comment:
Passing undef as a value is allowed and will be stored.  There a lots of examples if this in BackupInfo.pm.